honestcash-v2/src/app/modules/editor/components/editor/editor.component.ts

```diff
@@ -138,67 +122,104 @@ export class EditorComponent implements OnInit, OnDestroy {
-        editorToolsToLoad.push(
-          ...[this.scriptService.loadScript('https://cdn.jsdelivr.net/npm/@editorjs/header@2.2.4/dist/bundle.min.js'),
-            this.scriptService.loadScript('https://cdn.jsdelivr.net/npm/@editorjs/image@2.3.1/dist/bundle.min.js'),
-            this.scriptService.loadScript('https://cdn.jsdelivr.net/npm/@editorjs/embed@2.2.1/dist/bundle.min.js')]
-        );
-      }
+    if (this.isPlatformBrowser && !this.hasEditorInitStarted && !this.hasEditorInitialized) {
+      this.shouldEditorAllowTitleAndCustomElements = (
+          this.editingMode === EDITOR_EDITING_MODES.Write ||
+          this.editingMode === EDITOR_EDITING_MODES.Edit
+        ) &&
+        this.story &&
+        !this.story.parentPostId;
+      this.hasEditorInitStarted = true;
+      this.setupEditorScripts();
 
       this.scriptService.loadScript('https://cdn.jsdelivr.net/npm/@editorjs/editorjs@2.14.0/dist/editor.min.js').pipe(
         concatMap(() => forkJoin(
-          editorToolsToLoad
+          this.editorScripts
         ))
       ).subscribe(() => {
-        this.editorConfig = {
-          holder: 'editor',
-          initialBlock: 'paragraph',
-          onChange: this.onBodyChange.bind(this),
-          placeholder: this.editingMode !== EDITOR_EDITING_MODES.Comment ? 'Write your story...' : 'Write your comment...',
-          tools: {
-            paragraph: { // this is shared by all modes
-              class: Paragraph,
-              inlineToolbar: false,
-            },
-          }
-        };
-
-        if (this.editingMode === EDITOR_EDITING_MODES.Write || this.editingMode === EDITOR_EDITING_MODES.Edit) {
-          this.editorConfig.tools = {
-            ...this.editorConfig.tools,
-            header: {
-              class: Header,
-              inlineToolbar: false,
-            },
-            image: {
-              class: ImageTool,
-              inlineToolbar: false,
-              config: {
-                uploader: {
-                  uploadByFile: this.uploadImage.bind(this),
-                  uploadByUrl: this.downloadImageFromUrlAndUpload.bind(this)
-                }
-              }
-            },
-            embed: {
-              class: Embed,
-              inlineToolbar: false,
-            },
-          };
-        }
-
+        this.setupEditorConfig();
         this.editor = new EditorJS(this.editorConfig);
-        this.editor.isReady.then(() => {
-          this.store.dispatch(new EditorLoad());
-        });
+        this.setupEditorTitle();
       });
 
     }
   }
+
+  private onEditorReady() {
+    this.editorService.savePostLocally(this.story);
+    this.store.dispatch(new EditorLoad());
+    this.hasEditorInitialized = true;
+  }
+
+  private setupEditorScripts() {
+    this.editorScripts = [
+      this.scriptService.loadScript('https://cdn.jsdelivr.net/npm/@editorjs/paragraph@2.5.1/dist/bundle.min.js'),
```

Comment:
These need to be inside the repo as well so we don't get outside dependencies.

Reply:
added them locally the same way as simple-bitcoin-wallet